### ISSUE TITLE
no-member false positive for @property

### ISSUE THREAD

**Issue by novotl** [PyCQA/pylint#6708]
### Bug description

Contents of  `custom_property.py`:

```python
# pylint: disable=missing-class-docstring
# pylint: disable=missing-function-docstring
# pylint: disable=missing-module-docstring


def custom_prop(func):
    return property(func)


class MyClass:
    def __init__(self) -> None:
        self.text = "example_text"

    @property
    def one(self) -> str:
        return self.text + "_one"

    @custom_prop
    def two(self) -> str:
        return self.text + "_two"


MyClass().one.split("_")
MyClass().two.split("_")  # E1101: Method 'two' has no 'split' member (no-member)

# according to pylint this is correct, but fails at runtime `'str' object is not callable`
MyClass().two().split("_")

```

### Command used

```shell
pylint custom_property.py
```

### Pylint output

```shell
************* Module custom_property
custom_property.py:24:0: E1101: Method 'two' has no 'split' member (no-member)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 2.13.9
astroid 2.11.5
Python 3.9.7 (default, Feb 10 2022, 17:50:24)
[GCC 8.3.0]
```

**Comment by jacobtylerwalls** [comment]
Duplicate of #1127

**Comment by novotl** [comment]
Sorry, couldn't find the original 🙏
